api/valor_api/backend/core/datum.py

```diff
@@ -38,10 +41,39 @@ def create_datums(
     try:
         db.add_all(rows)
         db.commit()
-    except IntegrityError:
+    except IntegrityError as e:
         db.rollback()
-        # TODO: fix this exception
-        raise exceptions.DatumAlreadyExistsError("")
+        if "duplicate key value violates unique constraint" not in str(e):
+            raise e
+
+        # get existing datums
+        existing_datums: list[models.Datum] = []
+        for datum, dataset in zip(datums, datasets):
+            try:
+                existing_datums.append(fetch_datum(db, dataset.id, datum.uid))
+            except exceptions.DatumDoesNotExistError:
+                pass
+
+        if not ignore_existing_datums:
+            raise exceptions.DatumsAlreadyExistsError(
+                [datum.uid for datum in existing_datums]
+            )
```

Comment:
Instead of adding this flag could we instead cache all the datum uids that fail and then raise the exception after all the datums have been added?

This would remove the need for the extra param and still ensure that all new datums are ingested